Research/lva-node-app/app.js

```diff
@@ -0,0 +1,337 @@
+      let modules="";
+        for(let i=0; i<response.responseBody.length; i++)
+        {
+          modules+=response.responseBody[i].moduleId+" ";
+          console.log(response.responseBody[i].moduleId);
+        }
+        res.send("Successfully connected to your IoTHub. Found the following modules on your device: "+modules);
+    }, reason =>  //on error send error response
+    {
+      console.log(reason);
+      res.send("Unable to connect to device. Error response: "+ JSON.parse(reason.responseBody).Message);
+    });          
+}) 
+
+
+/**
+* update global variables when sent by user
+*/
+app.post('/globals', function (req, res)
+{
+  setGlobals(req, res);
+  res.send();
+})
+
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// defined functions
+
+/**
+* set global variables when passed by user 
+*/
+function setGlobals(req, res)
+{
+  for (let i in req.body)
+  {
+    let name=req.body[i].name;
+    switch(name)
+    {
+      case 'graphInstances':
+        graphInstances=req.body[i].value;
+        break;
+      case 'graphTopologies':
+        graphTopologies=req.body[i].value;
+        break;
+      case 'cameras':
+        cameras=req.body[i].value;
+        break;
+    }
+  }
+}
+
+/**
+* set device ID and IoTHub connection string
+* try to connect to the iot hub to make sure that the passed in values are valid!
+*/
+function setConfigs(req, res)
+{
+  DEVICE_ID=req.body["device-id"];
+  IOTHUB_CONNECTION_STRING=req.body["iothub-connection-string"];
+  convertIotHubToEventHubsConnectionString(IOTHUB_CONNECTION_STRING).then((eventhubString) =>
+  {
+    IOTHUB_ENDPOINT = eventhubString;
+  }).catch((error) =>
+  {
+    console.error(error);
+    res.send(error.message);
+  });
+}
+
+
+/** 
+* broadcast used for sending IoT Hub message data in a live stream 
+* @param {any} data - data to send through websocket
+*/
+wss.broadcast = (data) => 
+{
+    wss.clients.forEach((client) => 
+    {
+      if (client.readyState === WebSocket.OPEN) 
+      {
+        try 
+        {
+          client.send(data);
+        } catch (er) 
+        {
+          console.error(er);
+        }
+      }
+    });
+};
+
+/**
+* function that actually broadcasts hub messages to websocket
+*/
+async function receiveHubMessages() 
+  {
+      const consumerGroup = "$Default"; // name of the default consumer group
+      eventHubReader = new EventHubReader(IOTHUB_CONNECTION_STRING, consumerGroup);
+      (async () => 
+      {
+          await eventHubReader.startReadMessage((message, date, deviceId) => 
+          {
+            try 
+            {
+              const payload = 
+              {
+                IotData: message,
+                MessageDate: date || Date.now().toISOString(),
+                DeviceId: deviceId,
+              };
+
+              wss.broadcast(JSON.stringify(payload));
+            } catch (err) 
```

Comment:
this is the third variation of catch (err) i've seen in this file. :) lets be consistent on what we name the thing we're catching unless there's a reason not to be

Reply:
Good point! Fixed